src/Compilers/Core/Portable/SourceGeneration/SyntaxStore.cs

```diff
@@ -88,8 +91,16 @@ public IStateTable GetSyntaxInputTable(SyntaxInputNode syntaxInputNode, NodeStat
                             {
                                 try
                                 {
-                                    _cancellationToken.ThrowIfCancellationRequested();
-                                    syntaxInputBuilders[i].builder.VisitTree(root, state, model, _cancellationToken);
+                                    Stopwatch sw = Stopwatch.StartNew();
+                                    try
+                                    {
+                                        _cancellationToken.ThrowIfCancellationRequested();
+                                        syntaxInputBuilders[i].builder.VisitTree(root, state, model, _cancellationToken);
+                                    }
+                                    finally
+                                    {
+                                        _syntaxTimes[syntaxInputBuilders[i].node] = _syntaxTimes[syntaxInputBuilders[i].node].Add(sw.Elapsed);
```

Comment:
>_syntaxTimes[syntaxInputBuilders[i].node].Add(sw.Elapsed) [](http://example.com/codeflow?start=84&length=57)

Is there a reason to add rather than simply assigning `sw.Elapsed`?

Reply:
We're inside a double loop, so can sum up multiple elapsed times into a single result.